src/Components/Login.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+
+const uri = process.env.REACT_APP_API_ENDPOINT || 'http://localhost:8080'
+
+class Login extends Component {
+
+  state = {
+    username: '',
+    player: null
+  }
+
+  componentDidMount() {
+    this.loginStatus();
+  }
+
+  loginStatus() {
+    return fetch(`${uri}/login`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      credentials: 'include'
+    }).then(res => {
+      if(res.status < 400) return res.json()
```

Comment:
I don't know react/js too well, does `res.json()` need an await here? If not, do you know why not? I usually see json() conversion with an `await` call.

Reply:
`res.json()` returns a promise, so long as it's chained with another `.then()` the await isn't needed. 
But if I wanted to simplify it could refactor this to

```javascript
}).then(async res => {
  if(res.status >= 400) { throw res }
  let player = await res.json()
  this.setState({player})
  // ...
})
```

Reply:
Cool, thanks for the explanation! I think the chain is fine, whichever way you prefer to go do that :)